central/probeupload/service/service_impl.go

```diff
@@ -79,7 +80,7 @@ func (s *service) CustomRoutes() []routes.CustomRoute {
 		{
 			Route:         "/api/extensions/probeupload",
 			Authorizer:    user.With(permissions.Modify(resources.ProbeUpload)),
-			ServerHandler: http.HandlerFunc(s.handleProbeUpload),
+			ServerHandler: httputil.HandlerNotImplementedIfSettingDisabled(!env.DisableKernelPackageUpload.BooleanSetting(), http.HandlerFunc(s.handleProbeUpload)),
```

Comment:
Three negations here make the code hard to understand.
My suggestion:

 - Rename function to `httputil.DisableHandler(bool, func)`, a true disables the handler. Or `EnableHandler` which reacts on a positive setting.
 - Changing the env var to a positive `EnableKernelPackageUpload`

```suggestion
			ServerHandler: httputil.HandlerEnabled(env.EnableKernelPackageUpload.BooleanSetting(), http.HandlerFunc(s.handleProbeUpload)),
```

Reply:
I'm using the IfThenElse and with the switch to `EnableKernelPackageUpload` it'll become true enables it and false returns not implemented.